x/lockup/keeper/internal/events/emit.go

```diff
@@ -0,0 +1,110 @@
+package events
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/osmosis-labs/osmosis/v10/x/gamm/utils"
+	"github.com/osmosis-labs/osmosis/v10/x/lockup/types"
+)
+
+// EmitLockToken returns a new event when user lock tokens.
+func EmitLockToken(ctx sdk.Context, lock *types.PeriodLock) {
+	if ctx.EventManager() == nil {
+		return
+	}
+
+	ctx.EventManager().EmitEvents(sdk.Events{
+		lockTokenEvent(lock),
+	})
+}
+
+func lockTokenEvent(lock *types.PeriodLock) sdk.Event {
+	return sdk.NewEvent(
+		types.TypeEvtLockTokens,
+		sdk.NewAttribute(types.AttributePeriodLockID, utils.Uint64ToString(lock.ID)),
+		sdk.NewAttribute(types.AttributePeriodLockOwner, lock.Owner),
+		sdk.NewAttribute(types.AttributePeriodLockAmount, lock.Coins.String()),
+		sdk.NewAttribute(types.AttributePeriodLockDuration, lock.Duration.String()),
+		sdk.NewAttribute(types.AttributePeriodLockUnlockTime, lock.EndTime.String()),
+	)
+}
+
+// EmitLockToken returns a new event when user lock tokens.
+func EmitExtendLockToken(ctx sdk.Context, lock *types.PeriodLock) {
+	if ctx.EventManager() == nil {
+		return
+	}
```

Comment:
Why are we doing all these nil guards? Seems fine to me if it panics with NPE, this is misconfiguration that should get caught in testing, not be graceful no-op

Reply:
While I agree that no-op is probably not the way to go, I think it is worth having a check for nil event manager somewhere since `ctx` is injected into Osmosis's message server by its dependency - `cosmos-sdk`. I don't think we should be spending time on this right now though so it should be fine if we just remove the check